Terminology section: grammar edits, minor term edits leveraging existing RFCs. (#130)
ietf-wg-cats/draft-ietf-cats-framework: draft-ietf-cats-framework.md

This document makes use of the following terms:

Client:

> : An endpoint that requests service from a service provider network.

Author: Thanks reading and your feedback!

I think my confusion here (as a new comer) stems from my current assumption that CATS does not care about endpoints connected to a network (in general it does not interact with these endpoints) until they make a request to a CATS-backed service.  So this change narrows down the scope of devices considered a client from any connected device to only those requesting services that CATS devices must steer? Is this a bad assumption/logic?

Also, to me a host is a device that simply connects to a network as it is yet to enter into a client-server relationship? 

If these thoughts don't make sense or are irrelevant, happy to just remove the passive voice, thoughts on:
"An endpoint that connects to a service provider network."

Flow:

> : "A set of IP packets passing an Observation Point in the network during a certain time interval. All packets that belong to a particular Flow have a set of common properties derived from the data contained in the packet and from the packet treatment at the Observation Point" {{?RFC3954}}. Typically, CATS components identify flows by the 5-tuple transport coordinates (source address and destination address, source and destination port numbers, and protocol).

Reviewer: I would argue, we may not use 5-tuple, it can be 3-tuple, 4-tuple or even 6-tuple to identify. so making 5 as an example should be better.

"A logical grouping of packets during a time interval, which can be identified by some fields from the packet header, such as the 5-tuple transport coordinates (source address and destination address, source and destination port numbers, and protocol) "

Computing-Aware Traffic Steering (CATS):
 : A traffic engineering approach {{?RFC9522}} that takes into account the dynamic nature of computing resources (e.g., compute and storage) and network state to optimize service-specific traffic forwarding towards a given service contact instance. The CATS framework leverages various metrics to enable computing-aware traffic steering policies.

Metric:

>  : A quantitative measure that provides suitable input to a selection mechanism for CATS decision making.

Reviewer: LGTM

Computing metrics:

>   : Metrics specific to the computing resources in the underlying CATS system(s) as distinct from other metrics, such as network metrics. For further detail, see the set of metrics of computing computing metrics defined in {{?I-D.ietf-cats-metric-definition}}.

Reviewer 2: ```suggestion
  : Metrics specific to the computing resources in the underlying CATS system as distinct from other metrics, such as network metrics. For further detail, see the set of computing computing metrics defined in {{?I-D.ietf-cats-metric-definition}}.
```

Author: Thanks for catching this typo! I fixed it with the last commit.

Service:

>   : An interface a service provider exposes to a client to perform a function. Service providers establish services by orchestrating a set of resources (networking, compute, storage, etc.).

Reviewer 2: I have a preference for the OLD.

Reviewer 2: > Just want to be sure, were the changes to line 153 good to go? I did not revert these. OLD (above in code snippet). NEW: "The service provider retains control of internal resources and the service logic. For example, these resources may be:"

That one is OK. Thanks

    * Exposed by one or multiple processes.
    * Provided by virtual instances, physical, or a combination thereof.
    * Hosted within the same or distinct nodes.
    * Hosted within the same or multiple service sites.
    * Chained to provide a service using a variety of means.

  : How a service provider structures its services remains out of the scope of CATS.
  : Service providers may provide the same service in many locations; each of them constitutes a service instance.

Computing Service:
  : A service offered to a client by a service provider by orchestrating a set of computing resources.

CATS Service ID (CS-ID):
 : An identifier representing a service, which the clients use to access it. See {{cats-ids}}.

Service instance:
  : An instance of running resources according to a given service logic.
  : A service provider may enable many service instances that adhere to the same service logic to provide the same service.
  : A service instance runs in a service site and one or more instances may service clients’ requests.

Service site:
 : A location that hosts the resources that implement an offered service.
 : A service site may be a node or a set of nodes.

>  : A CATS-serviced site is a service site that a service provider connects to a CATS-Forwarder.

Reviewer 2: I'm not sure this change is needed as this may infer who controls what, etc. I would keep the OLD.

Author: Thanks for you thoughts here, I was not sure on this one (as commented above). 

To simply address passive voice, how is: "A CATS-serviced site is a service site connected to a CATS-Forwarder."

Out of curiosity, do you know if these details (who does what) will get hashed out in a later document (as in one attempting to specify a CAT protocol)?

Service contact instance:
  : A client-facing service function instance that is responsible for receiving requests in the context of a given service.
  : A service contact instance can handle one or more service instances.

>   : Steering beyond a service contact instance remains transparent to both clients and CATS components.

Reviewer 2: "is hidden" reflects better the intent. I would leave the OLD.

Author: Reverted!

  : A service contact instance processes a client's service request according to the service logic (e.g., handle locally or solicit backend resources).

>   : A service contact instance must be reachable via at least one Egress CATS-Forwarder.

Reviewer 2: ```suggestion
  : A service contact instance is reachable via at least one Egress CATS-Forwarder.
```

  : Clients may access a service via multiple service contact instances running at the same or different locations (service sites).
  : A service contact instance may dispatch service requests to one or more service instances (e.g., a service contact instance that behaves as a service load-balancer).

CATS Service Contact Instance ID (CSCI-ID):
 : An identifier of a specific service contact instance. See {{cats-ids}}.

Service request:
 : A request to access or invoke a specific service. CATS-Forwarders steer a service request to a service contact instance.
 : Clients must place a service request using service-specific protocols.

>  : Clients direct service requests to a CS-ID without explicit knowledge of CATS-Forwarders.

Reviewer: to a CS-ID? This sentence highlight that the destination is the service provider address, not the CATS forwarder.

Clients direct service requests to a server (identified by a CS-ID),  without explicit knowledge of CATS-Forwarders.
may be better?

CATS-Forwarder:

>  : A network entity that steers traffic specific to a service request towards a corresponding yet-selected service contact instance according to provisioned forwarding decisions.  A C-PS, which may or may not be on the CATS-Forwarder, makes forwarding decisions.

Reviewer 2: The OLD reflect better the intent; C-PS computes paths, but actual forwarding decision are implemented/made by a CATS forwarder.

 : A CATS-Forwarder may behave as an Ingress or Egress CATS-Forwarder.

Ingress CATS-Forwarder:
 : An entity that steers service-specific traffic along a CATS-computed path that leads to an Egress CATS-Forwarder that connects to the most suitable service site that hosts the service contact instance selected to satisfy the initial service request.

Egress CATS-Forwarder:
: An entity located at the end of a CATS-computed path which connects to a CATS-serviced site.

CATS Path Selector (C-PS):
 : A functional entity that selects paths towards service locations and instances in order to accommodate the requirements of service requests. The path selection engine takes into account the service and network status information. See {{sec-cps}}.

CATS Service Metric Agent (C-SMA):
 : A functional entity responsible for collecting service capabilities and status, and for reporting them to a CATS Path Selector (C-PS). See {{sec-csma}}.

CATS Network Metric Agent (C-NMA):

>  : A functional entity responsible for collecting network capabilities and status, and for reporting them to a C-PS. See {{sec-cnma}}.

Reviewer: same as above.

CATS Traffic Classifier (C-TC):

>  : A functional entity responsible for determining which packets belong to a traffic flow for a specific service request and forwarding a flow's packets along a C-PS computed path that leads to the relevant service contact instance. See {{sec-ctc}}.

Reviewer: OLD one may be better. This sentence is too long, making it hard to understand
